submission_runner.py

```diff
@@ -197,6 +193,27 @@ def train_once(
 ) -> Tuple[spec.Timing, spec.Steps]:
   data_rng, opt_init_rng, model_init_rng, rng = prng.split(rng, 4)
 
+  # Logger setup.
+  logging.info('Initializing logger.')
+  metrics_logger = None
+  if log_dir is not None:
+    hparams_fname = os.path.join(log_dir, 'hparams.json')
+    meta_data = get_meta_data(workload)
+    meta_fname = os.path.join(log_dir, 'meta_data.json')
```

Comment:
nit: I prefer `filename` instead of `fname` because the latter can imply function or something else :)